app/javascript/src/utils/compiler.js

```diff
@@ -410,3 +411,101 @@ function evaluateForLoops(joinedItems) {
 
   return joinedItems
 }
+
+function parseArrayValues(input) {
+  const commaRegex = /, */g
+
+  const result = []
+  let commaMatch
+  let nextStartingIndex = 0
+  let lastValidCommaEndIndex = -1
+  while ((commaMatch = commaRegex.exec(input)) != null) {
+    // Check if the comma is inside parentheses (e.g. the second comma in "[1, (2, 3), 4]")
+    // because the parenthesis group should be taken as one value (e.g. for the previous
+    // example, we should return ["1", "(2, 3)", "4"], not ["1", "(2", "3)", "4"])
+    const openBracketIndex = input.indexOf("(", nextStartingIndex)
+    if (openBracketIndex >= 0 && openBracketIndex < commaMatch.index) {
+      const closingBracketIndex = getClosingBracket(input, "(", ")", openBracketIndex - 1)
+      nextStartingIndex = closingBracketIndex < 0 ? input.length : closingBracketIndex
+    } else {
+      const commaEndIndex = commaMatch.index + commaMatch[0].length - 1
+      result.push(input.substring(lastValidCommaEndIndex + 1, commaMatch.index))
+
+      lastValidCommaEndIndex = commaEndIndex
+      nextStartingIndex = lastValidCommaEndIndex + 1
+    }
+
+    commaRegex.lastIndex = nextStartingIndex
+  }
+
+  const lastValue = input.substring(lastValidCommaEndIndex + 1)
+  if (lastValue.length > 0) {
+    result.push(lastValue)
+  }
+  return result
+}
+
+function evaluateEachLoops(joinedItems) {
+  const eachRegex = /@each\s*\((\w+)(?:,\s+(\w+))?\s+in\s+((?:.|\n)+?)\s*\)\s*\{/g
+
+  let match
+  while ((match = eachRegex.exec(joinedItems)) != null) {
+    const [_, valueVar, indexVar, iterableStr] = match
+
+    let iterable = []
+    if (iterableStr[0] === "[" && iterableStr[iterableStr.length - 1] === "]") {
+      iterable = parseArrayValues(iterableStr.substring(1, iterableStr.length - 1))
+    } else {
+      const language = get(defaultLanguage)
+      const constants = get(workshopConstants)
+
+      const sanitizedIterableStr = iterableStr.toLowerCase().trim()
+
+      // TODO: support constant names in other languages
```

Comment:
Other languages won't be supported, they aren't supported anywhere else either.

Reply:
Fair enough. But what if they were? 🤔

Ok ok, I will remove the TODO.

Reply:
Maybe one day when someone actually asks for it. But for now the amount of maintenance required far outweighs the benefits